### ISSUE TITLE
64 bit Windows only?

### ISSUE THREAD

**Issue by cambragol** [Fenyx4/U4DosRandomizer#5]
Tried to get the alpha working, but I am assuming it is a 64bit app only, as I could not start it in XP or Dos.

Just curious.

**Comment by Fenyx4** [comment]
Thought I could get away with only doing 64bit in the prerelease. My bad.

I just added x86 and linux-x64 zips to the release. Let me know if those work better.
I did try the x86 one in DOS and it didn't work but it should in XP I hope.

Thanks!

**Comment by cambragol** [comment]
I just tried the new x86 version in windows XP, and I am still getting a 'this is not a valid Win32 application' error.

**Comment by Fenyx4** [comment]
Dangit. Now I am regretting writing this in .Net core.

I'll mess around with it this weekend and see if I can get it building in a version of .Net Framework that will hopefully work on XP.

**Comment by Fenyx4** [comment]
[U4DosRandomizer-net35.zip](https://github.com/Fenyx4/U4DosRandomizer/files/5111598/U4DosRandomizer-net35.zip)

Can you try this one? If it doesn't work you may have to install [Microsoft .NET Framework 3.5](https://www.microsoft.com/en-us/download/details.aspx?id=21). If it still doesn't work then I'll have to go back to the drawing board.

**Comment by cambragol** [comment]
Gave it a shot, with the dll to, but now it gives a 'the application failed to initialize properly 0c0000135' error.

Please don't make work for yourself to make it work on my antiquated hardware.

I run XP in a virtual machine. My normal machine is MacOS Mojave. 
Maybe a Dos version of the randomizer would be best, as anyone running U4 will have a Dos environment setup, regardless their actual OS.

**Comment by Fenyx4** [comment]
I added and osx-x64. Maybe that'll work?

I considered writing it so it ran in DOS when I first started working on it. But thought it might be more accessible if it ran outside the emulator, maybe I could extend it to handle other ports someday (like SMS) and having it run outside might make that easier and I just wanted to write in something more modern to make things easier on me. :)

**Comment by cambragol** [comment]
I installed Microsoft .NET Framework 3.5, but it still did not work.

Just tried the osx-x64 version. That actually starts, but I get an error as it can't find the path to 'World.MAP'. It is looking in /Users/Admin/World.map
So maybe the paths are hardcoded?
Anyways, hope I am helping out here and not making work. Just trying to give the Alpha a try.

**Comment by Fenyx4** [comment]
Yay starting! 
I guess I'm not sure how Macs do working directories. Was the executable, U4DosRandomizer, in /Users/Admin or was it in the directory with World.MAP?

You could also try 'U4DosRandomizer -p <path to Ultima IV installation directory>'.

If it fails to find World.MAP in the current working directory do you think it would be more useful for it to prompt for a path or to just include -p as an option in the error message?

Definitely helping! Thank you!

**Comment by cambragol** [comment]
The executable was in the U4 folder, with World.MAP. If I put a copy of World.MAP in /Users/Admin, u4DosRandomizer finds it, moves forward, then fails when it can't find the next file, Avatar.exe

Prompting for a path would likely fix it, if the path can't be found automatically.

Tried the -p flag, but I can't run it from the command line. 

Hopefully that sheds a little more light.

**Comment by Fenyx4** [comment]
I just pushed a new release. https://github.com/Fenyx4/U4DosRandomizer/releases/tag/v0.1-alpha.1

It _should_ prompt you for a path if it doesn't find the U4 files in the current directory. I don't have a mac so couldn't test it. Worked on my machine though. :)

Hopefully that works for you. 🤞

**Comment by cambragol** [comment]
Got it halfway working, after a path hiccup (dragging the path to the console leaves a space at the end which caused the prog to choke)

However, it gets stuck here:

<img width="572" alt="Screen Shot 2020-08-24 at 14 53 02" src="https://user-images.githubusercontent.com/5019739/91008606-b1378c80-e619-11ea-8646-6a51c5788fce.png">

**Comment by Fenyx4** [comment]
I think I see the problem. It found cove.tlk but it isn't the original file so it checks cove.tlk.orig and fails to find that.

Is your cove.tlk patched or something?

I will try to make a more useful error message.

**Comment by cambragol** [comment]
Hmm...that is weird. I checked it twice, and I pulled the files directly from the .dmg file of the GOG version, I believe. So it shouldn't be patched. But looking in the original folder there is a list of fixes, and cove is one of them. It is version 1.01. of Ultima IV.

The fixes are listed here:

- Charm in Cove asks her question
- Water in Lord British's castle asks its question
- Alkerion in Minoc asks his question
- Shamino in Skara Brae asks his question
- fixed mispelling of keyword of prisoner in Yew
- fixed incorrect guard indices in Serpents Hold

Which indicates several .tlk files may cause hiccups for your randomizer.

**Comment by Fenyx4** [comment]
Got another release for you!
https://github.com/Fenyx4/U4DosRandomizer/releases/tag/v0.1-alpha.2

This should fix the TLK problem. At least it worked when I grabbed the version 1.01 from http://www.pixsoriginadventures.co.uk/category/ultima-patcher/.

Really hopes this works for you this time!

(Also it now adds bridges.)

**Comment by cambragol** [comment]
Okay, tried it out.

Got a little further along this time. But then it choked:

<img width="606" alt="Screen Shot 2020-08-26 at 17 44 23" src="https://user-images.githubusercontent.com/5019739/91282173-f5ff2700-e7c3-11ea-99de-55c86fbef0ac.png">

I'll try it with the version you linked. Maybe it is system related.

**Comment by Fenyx4** [comment]
Okay! I have high hopes this time!

https://github.com/Fenyx4/U4DosRandomizer/releases/tag/v0.1-alpha.3

Creating a mini bitmap of the map is the last thing it does and that is what you crashed on. I was able to reproduce the error in Linux. It could be solved by installing some other libraries in the OS itself but I don't want people to have to do that.
So I replaced the System.Draw calls with a different library and that got it working in Linux.

Give it a try! 🤞
